### PR TITLE
#128 Use GitHub Actions instead of Heroku to run periodic builds of the advisories file

### PR DESCRIPTION
**Pull request by slash3b**
Implements Github workflow that builds project hourly and then pushes to the
`Roave\SecurityAdvisories` repository.

In main repository settings it is necessary to add "secret" under name `TOKEN` which value should be a personal access tokens with `write:packages` permission.

### CODE REVIEW

**Comment by slash3b** [comment]
@Ocramius could you take a look at this please?
It keeps building master branch https://travis-ci.org/github/Roave/SecurityAdvisoriesBuilder/builds/697343890
Did something change ? @_@

**Comment by Ocramius** [comment]
Haven't looked at it yet, but I shall 👍

File: .github/workflows/scheduled_build.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Hourly build
+
+on:
+  schedule:
+    - cron: '0 * * * *'
+
+jobs:
+  build:
+    if: (github.event_name == 'schedule' && github.repository == 'Roave/SecurityAdvisoriesBuilder') || (github.event_name != 'schedule')
```
Comment (Ocramius) [inline .github/workflows/scheduled_build.yml:9]: `&& github.repository == 'Roave/SecurityAdvisoriesBuilder'` is this still needed at all?

File: .github/workflows/scheduled_build.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Hourly build
+
+on:
+  schedule:
+    - cron: '0 * * * *'
```
Comment (Ocramius) [inline .github/workflows/scheduled_build.yml:5]: I never remember cron syntax: is this "hourly"?

File: .github/workflows/scheduled_build.yml (hunk 1 above)
Comment (Ocramius) [reply .github/workflows/scheduled_build.yml:9]: Also: why are other events considered? Maybe we want to only allow a manual trigger?

File: .github/workflows/scheduled_build.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v2
+        with:
+          path: vendor
+          key: ${{ runner.os }}-node-${{ hashFiles('**/composer.lock') }}
```
Comment (Ocramius) [inline .github/workflows/scheduled_build.yml:20]: We only have one OS, so I think the key is just based on the `composer.lock` hash (no `**` needed either)

File: .github/workflows/scheduled_build.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        id: dep-install
+        if: steps.composer-cache.outputs.cache-hit != 'true'
```
Comment (Ocramius) [inline .github/workflows/scheduled_build.yml:26]: I think the `composer install` should be performed regardless

File: .github/workflows/scheduled_build.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        run: composer install --prefer-dist --no-progress --no-suggest
```
Comment (Ocramius) [inline .github/workflows/scheduled_build.yml:27]: `--prefer-dist` can be skipped here :+1:

File: .github/workflows/scheduled_build.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Build advisories
+        id: build
+        env:
+          GITHUB_TOKEN: ${{ secrets.TOKEN }}
+        run: php build-conflicts.php
+
+      - name: Push to "Roave/SecurityAdvisories"
+        run: |
+          cd build/roave-security-advisories
+          git push origin master
```
Comment (Ocramius) [inline .github/workflows/scheduled_build.yml:47]: Excellent! \o/

**Comment by Ocramius** [comment]
> @Ocramius could you take a look at this please?
 > It keeps building master branch https://travis-ci.org/github/Roave/SecurityAdvisoriesBuilder/builds/697343890
 > Did something change ? @_@

Don't worry about this: we'll kill travis-ci very soon :-)

**Comment by slash3b** [comment]
> > @Ocramius could you take a look at this please?
> > It keeps building master branch https://travis-ci.org/github/Roave/SecurityAdvisoriesBuilder/builds/697343890
> > Did something change ? @_@
> 
> Don't worry about this: we'll kill travis-ci very soon :-)

I'd like to help.

File: .github/workflows/scheduled_build.yml (hunk 1 above)
Comment (Ocramius) [reply .github/workflows/scheduled_build.yml]: Thanks for clarifying!

I think we should note this in a comment in here, to make the conditional clear.

File: .github/workflows/scheduled_build.yml (hunk 2 above)
Comment (Ocramius) [reply .github/workflows/scheduled_build.yml:5]: Thanks!

**Comment by Ocramius** [comment]
Thanks @slash3b! 💪

File: .github/workflows/scheduled_build.yml (hunk 1 above)
Comment (slash3b) [reply .github/workflows/scheduled_build.yml]: I'm going to add it in next PR

**Comment by slash3b** [comment]
Thank you @Ocramius , I'm really glad I can be of any help ^^

**Comment by Ocramius** [comment]
@slash3b it's really helping, thanks! :100:

**Pull request merged**